cache.go

```diff
@@ -31,13 +33,15 @@ type namekey struct {
 // Cache supports patsy.Dir and patsy.Path, but cached so they can be used in
 // tight loops without hammering the filesystem.
 type Cache struct {
-	env    vos.Env
-	dirsm  *sync.RWMutex
-	pathsm *sync.RWMutex
-	namesm *sync.RWMutex
-	dirs   map[string]string
-	paths  map[string]string
-	names  map[namekey]string
+	env       vos.Env
+	dirm      *sync.RWMutex
+	dirsm      *sync.RWMutex
+	pathm     *sync.RWMutex
+	namem     *sync.RWMutex
+	dirCache  map[string]string
+	dirsCache map[string]map[string]string
+	pathCache map[string]string
+	nameCache map[namekey]string
```

Comment:
I removed the plural `s` suffix from all the vars because it was confusing with adding `dirs`, though I'm happy to instead come up with a different name for the `dirs` cache and mutex.